services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -149,30 +149,16 @@ private void loadAllPushes(List<OfflinePushStatus> offlinePushStatusList) {
     pushStatusCollector.start();
     try (AutoCloseableLock ignore = clusterLockManager.createClusterWriteLock()) {
       LOGGER.info("Load all pushes started for cluster {}'s {}", clusterName, getClass().getSimpleName());
-      // Subscribe to changes first
-      List<OfflinePushStatus> refreshedOfflinePushStatusList = new ArrayList<>();
       for (OfflinePushStatus offlinePushStatus: offlinePushStatusList) {
         try {
+          topicToPushMap.put(offlinePushStatus.getKafkaTopic(), offlinePushStatus);
           routingDataRepository.subscribeRoutingDataChange(offlinePushStatus.getKafkaTopic(), this);
-
+          getOfflinePushAccessor().subscribePartitionStatusChange(offlinePushStatus, this);
```

Comment:
I am trying to understand where the race comes from. The `change` mentioned in the description, was it come from the `onPartitionStatusChange` code path? In that function, we seem to need to acquire a cluster level read lock, so supposedly, the change would be blocked as this code block is already holding a cluster write lock, isn't it? or the change is from some other code path that I misunderstood?

Reply:
Yeah, so it is not like this -
In the original code, we load OfflinePushStatus from ZK first for everyone in a loop and then we subscribe to change one by one, apparently, if there is any update to the ZNODE in between, they should get ignored.

Reply:
Got it. Thanks for the explanation!